mutagen/dsf.py

```diff
@@ -0,0 +1,314 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2017  Boris Pruessmann
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+
+"""Read and write DSF audio stream information and tags."""
+
+import sys
+import struct
+
+from ._compat import cBytesIO, reraise, endswith
+
+from mutagen import FileType, StreamInfo
+from mutagen._util import cdata, MutagenError, loadfile, convert_error
+from mutagen.id3 import ID3
+from mutagen.id3._util import ID3NoHeaderError, error as ID3Error
+
+
+__all__ = ["DSF", "Open", "delete"]
+
+
+class error(MutagenError):
+    pass
+
+
+class DSFHeaderError(error):
```

Comment:
This was inspired by what I found in flac.py. I believe you are asking whether there is value in differentiating between general exceptions and header exceptions - which there probably isn't. Will rework.